### PR TITLE
Add assertSame and assertNotSame methods.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
@ilya-g

File: libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt [hunk 1]
```diff
@@ -10,6 +10,12 @@ class BasicAssertionsTest {
     }
 
     @Test
+    fun testAssertSame() {
```
Comment (piotrek1543) [inline libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt]: `test` prefix reminds me old days of `JUnit`, where it was obligatory to run any test :-)

Could you tell me why there is doubled `kotlin` and `test` packages in path: `kotlin/libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/`? Just wondering.

Comment (ilya-g) [reply libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt]: `kotlin.test/common` is the project directory, then `src/test/kotlin` is the source root for tests, then `kotlin/test/tests` is the package name for tests of `kotlin.test` :)

**Comment by ilya-g** [comment]
I've merged it with minor edits: https://github.com/JetBrains/kotlin/compare/485ca10~2...485ca10

**Comment by JakeWharton** [comment]
Thanks!

**Pull request closed without merging**